packages/expo-video-thumbnails/android/src/main/java/expo/modules/videothumbnails/VideoThumbnailsModule.kt

```diff
@@ -0,0 +1,115 @@
+package expo.modules.videothumbnails
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.media.MediaMetadataRetriever
+import android.net.Uri
+import android.util.Log
+import android.webkit.URLUtil
+import expo.modules.core.errors.ModuleDestroyedException
+import expo.modules.core.utilities.FileUtilities
+import expo.modules.interfaces.filesystem.Permission
+import expo.modules.kotlin.Promise
+import expo.modules.kotlin.exception.CodedException
+import expo.modules.kotlin.exception.Exceptions
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.OutputStream
+
+class VideoThumbnailsModule : Module() {
+  private val context
+    get() = appContext.reactContext ?: throw Exceptions.ReactContextLost()
+  private val moduleCoroutineScope = CoroutineScope(Dispatchers.IO)
+
+  override fun definition() = ModuleDefinition {
+    Name("ExpoVideoThumbnails")
+
+    AsyncFunction("getThumbnail") { sourceFilename: String, options: VideoThumbnailOptions, promise: Promise ->
+      if (URLUtil.isFileUrl(sourceFilename) && !isAllowedToRead(Uri.decode(sourceFilename).replace("file://", ""))) {
+        throw ThumbnailException()
+      }
+
+      withModuleScope(promise) {
+        val thumbnail = GetThumbnail(sourceFilename, options, context).execute()
+          ?: throw GenerateThumbnailException()
+
+        try {
+          val path = FileUtilities.generateOutputPath(context.cacheDir, "VideoThumbnails", "jpg")
+          val outputStream: OutputStream = FileOutputStream(path)
```

Comment:
The `use` function should handle the output stream. It uses the `try-with-resource` mechanism known from Java. So you don't have to close and flush the stream. 

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html